packages/core/src/scripts/migrate-oz-cli-project.ts

```diff
@@ -0,0 +1,426 @@
+
+  for (const manifestFile of developmentManifests) {
+    console.log(SUCESS_CHECK + `Deleting unused development manifest ${manifestFile}`);
+    await fs.unlink(manifestFile);
+  }
+
+  console.log(SUCESS_CHECK + `Deleting ${PROJECT_FILE}`);
+  await fs.unlink(PROJECT_FILE);
+}
+
+async function migrateManifests(oldManifests: string[]): Promise<MigrationOutput> {
+  const manifestsMigrationData: Record<string, unknown> = {};
+  const newManifests: Record<string, ManifestData> = {};
+
+  for (const manifestFile of oldManifests) {
+    const oldManifest = JSON.parse(await fs.readFile(manifestFile, 'utf8'));
+
+    if (oldManifest.manifestVersion === undefined) {
+      throw new Error('Migration failed: manifest version too old');
+    }
+
+    const [major, minor] = oldManifest.manifestVersion.split('.');
+
+    if (major === '3') {
+      continue;
+    }
+
+    // 2.2 latest version
+    if (major !== '2' || minor !== '2') {
+      throw new Error(`Migration failed: expected manifest version 2.2, got ${major}.${minor} instead`);
+    }
+
+    const network = getNetworkName(manifestFile);
+    newManifests[manifestFile] = updateManifest(oldManifest);
+    manifestsMigrationData[network] = getExportData(oldManifest);
+  }
+
+  return {
+    newManifests,
+    manifestsMigrationData,
+  };
+}
+
+async function getManifests(): Promise<string[]> {
+  const files = await fs.readdir(OPEN_ZEPPELIN_FOLDER);
+  return files.filter(isManifestFile).map(location => path.join(OPEN_ZEPPELIN_FOLDER, location));
+}
+
+async function getProjectFile(): Promise<LegacyProjectFile> {
+  return JSON.parse(await fs.readFile(PROJECT_FILE, 'utf8'));
+}
+
+function isManifestFile(fileName: string): boolean {
+  const network = getNetworkName(fileName);
+  return isPublicNetwork(network) || isDevelopmentNetwork(network) || isUnknownNetwork(network);
+}
+
+function getNetworkName(fileName: string): string {
+  return path.basename(fileName, '.json');
+}
+
+function isDevelopmentNetwork(network: string): boolean {
+  // 31337      => buidler evm
+  // 1337       => ganache
+  // 13+ digits => ganache also uses timestamps
+  return /^dev-(3?1337|\d{13,})$/.test(network);
+}
+
+function isUnknownNetwork(network: string): boolean {
+  return !isDevelopmentNetwork(network) && /^dev-\d+$/.test(network);
+}
+
+function isPublicNetwork(network: string): boolean {
+  return ['mainnet', 'rinkeby', 'ropsten', 'kovan', 'goerli'].includes(network);
+}
+
+function getNewManifestLocation(oldName: string): string {
+  return isUnknownNetwork(oldName) ? oldName.replace('dev', 'unknown') : oldName;
+}
+
+async function writeJSONFile(location: string, data: unknown): Promise<void> {
+  await fs.writeFile(location, JSON.stringify(data, null, 2));
+}
+
+function getExportData(oldManifest: LegacyManifest): ExportData {
+  // we flexibilize the typing to allow required fields deletion
+  const manifestsMigrationData: Partial<LegacyManifest> = oldManifest;
```

Comment:
We need to create a copy to avoid modifying the argument. (In this sense, TypeScript is right to disallow deletion!)

```suggestion
  const manifestsMigrationData: Partial<LegacyManifest> = Object.assign({}, oldManifest);
```

Additionally, here's how to properly type this variable to be able to delete properties and return it without the unsafe `as` cast. (Not a suggestion because the cast should be removed too.)

```typescript
  const manifestsMigrationData: ExportData & Partial<LegacyManifest> = ...;
```

Reply:
Shouldn't we then deep copy instead of `Object.assign()`?

Reply:
By "avoid modifying the argument" I was referring to the `delete` operations that are also contained in this function. I think we can establish the convention that objects aren't modified unless they are meant to be, and if so it's not necessary to deep copy.